### DISCUSSION TITLE
GitHub would shut off Probot DCO App in approximately 4 weeks

### DISCUSSION THREAD

**Discussion by lloydchang** [open-gitops/project#27] [General]
@todaywasawesome @scottrigby @murillodigital 
FYI: GitHub would shut off Probot DCO App in approximately 4 weeks
Details at:
1. https://github.com/probot/dco/issues/162#issuecomment-941149056
2. https://github.com/probot/dco/issues/162#issuecomment-941165340
3. https://github.com/probot/dco/issues/162#issuecomment-941224634
4. https://github.com/probot/dco/issues/162#issuecomment-941236942
5. https://github.com/openjs-foundation/cross-project-council/issues/632#issuecomment-941307810
6. https://github.com/probot/dco/issues/162#issuecomment-941498012
7. https://github.com/probot/dco/issues/162#issuecomment-941546270

Thank you!

**Comment by lloydchang** [comment]
Tangentially, I just copied https://github.com/gitops-working-group/gitops-working-group/discussions/174 discussion to this https://github.com/open-gitops/project/discussions/27 in alignment with the moving plan described at https://github.com/cncf/tag-app-delivery/pull/146#issuecomment-943600301

**Comment by scottrigby** [comment]
Until there's a new standard tool, I like what I see from the keptn folks implementation https://github.com/probot/dco/issues/162#issuecomment-943267084

@mowies how has your dco gh action been working out for you all since implementing it? Any issues?

**Reply by mowies** [reply]
Hi @scottrigby !
So, since this whole thing went does with the DCO GH App not responding anymore, a few things happened:
- the keptn project looked for an alternative to the GH app to still have some DCO check since it is obligatory for CNCF projects
- I implemented a small pipeline under the keptn GH org which takes a different dco checker tool and uses it in all the keptn PRs. you can find that in [this repo](https://github.com/keptn/gh-automation)
- that pipeline then got donated to the CNCF and now also lives [here](https://github.com/cncf/dcochecker)
- In keptn, we left both my new pipeline and the DCO GH app running just to see whether the DCO app would actually be shut down in the end.

The current state of DCO in the keptn project:
- We are still using both DCO checkers now
- The DCO App is still online and can be used normally
- You can also use our separate DCO pipeline or the CNCF donated one at any time if you want to
- In the keptn project, we plan to try out EasyCLA instead of DCO in the future but no timeline or results yet
- We had one issue with the custom implemented DCO pipeline:
  - When a dev branch (dev branch 1) is used as base for another dev branch (dev branch 2) which is the merged back into the dev branch 1, the DCO pipeline will then complain in the dev branch 1 PR that the DCO is not met because there are some committer mail address mismatches. It can't handle the GitHub no-reply mail addresses that are used in commits after a merge if the committer has some specific privacy settings around their mail address on GitHub.
  - Example: https://github.com/keptn/keptn/pull/6461 with the failing DCO check: https://github.com/keptn/keptn/runs/4702772622?check_suite_focus=true

**Reply by scottrigby** [reply]
@mowies Thank you! 👏🙇
